include/common/firmware/i2c.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "stm32g4xx_hal_conf.h"
+
+
+#ifndef __I2C_H__
+#define __I2C_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+struct I2CConfig {
+    I2C_TypeDef* instance;
+    uint8_t address;
+    uint16_t address_mode;
+};
+I2C_HandleTypeDef MX_I2C_Init(struct I2CConfig* conf);
+
+#define REGISTER1 I2C1
```

Comment:
This, I don't think is a good idea. using #define to point to other defines tends to muddy things up a lot, and is really only worth it to kind of move the same number between entirely separate semantic concepts. For instance, if this was `eeprom.h` and this was `#define EEPROM_I2C I2C1`, that would be one thing. But here, 
- `REGISTER1` is actually _less_ specific a name than `I2C1`
- You have to include this file to get `REGISTER1`, which means you're transitively including the hal, so you're not really gaining anything out of it